pkg/middleware/validation.go

```diff
@@ -45,31 +48,50 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(valid ValidationResponse, r *http.Request, mod string, version string) {
+	if len(valid.Message) > 0 {
+		entry := log.EntryFromContext(r.Context())
+		entry.Warnf("Module %s@%s %s", mod, version, valid.Message)
+	}
+}
+
 type validationParams struct {
 	Module  string
 	Version string
 }
 
-func validate(hook, mod, ver string) (bool, error) {
+type ValidationResponse struct {
+	Status  int
+	Message []byte
+}
+
+func validate(hook, mod, ver string) (ValidationResponse, error) {
 	const op errors.Op = "actions.validate"
 
 	toVal := &validationParams{mod, ver}
 	jsonVal, err := json.Marshal(toVal)
 	if err != nil {
-		return false, errors.E(op, err)
+		return ValidationResponse{Status: http.StatusForbidden}, errors.E(op, err)
 	}
 
 	resp, err := http.Post(hook, "application/json", bytes.NewBuffer(jsonVal))
 	if err != nil {
-		return false, errors.E(op, err)
+		return ValidationResponse{Status: http.StatusForbidden}, errors.E(op, err)
 	}
 
 	switch {
 	case resp.StatusCode == http.StatusOK:
-		return true, nil
+		return maybeReadResponseReason(resp), nil
```

Comment:
This switch statement can be factored out to read a bit better: 

```
switch resp.Status {
  case http.StatusOK:
    return maybeReadResponseReason(true), nil
  case http.StatusForbidden:
    return maybeReadResponseReason(false), nil
  default:
    return ValidationResponse{Status: http.StatusForbidden}, errors.E(op, "Unexpected status code ", resp.StatusCode)
}
```

While maybeReadResponseReason can just take a bool to populate `Valid` from above

Reply:
The response is required to be passed through in order to fill in the message parsed from the response.Body https://github.com/gomods/athens/blob/933d4430dadc1cfd1b8a34d85162930e503434b3/pkg/middleware/validation.go#L95

Reply:
made the changes to switch statement https://github.com/gomods/athens/pull/1517/commits/12dba10897b22c735db29b02d50fd64a85bba2ea

Reply:
Ah right, thanks!